convert_hf_to_gguf.py

```diff
@@ -93,11 +100,38 @@ def __init__(self, dir_model: Path, ftype: gguf.LlamaFileType, fname_out: Path,
             else:
                 logger.info(f"choosing --outtype bf16 from first tensor type ({first_tensor.dtype})")
                 self.ftype = gguf.LlamaFileType.MOSTLY_BF16
-        ftype_up: str = self.ftype.name.partition("_")[2].upper()
-        ftype_lw: str = ftype_up.lower()
-        # allow templating the file name with the output ftype, useful with the "auto" ftype
-        self.fname_out = fname_out.parent / fname_out.name.format(ftype_lw, outtype=ftype_lw, ftype=ftype_lw, OUTTYPE=ftype_up, FTYPE=ftype_up)
-        self.gguf_writer = gguf.GGUFWriter(path=None, arch=gguf.MODEL_ARCH_NAMES[self.model_arch], endianess=self.endianess, use_temp_file=self.use_temp_file,
+
+        # Fallback to model directory name if metadata name is still missing
+        if self.metadata.name is None:
+            self.metadata.name = dir_model.name
+
+        # Generate parameter weight class (useful for leader boards) if not yet determined
+        if self.metadata.size_label is None:
+            expert_count = self.hparams.get("num_local_experts", 0)
+            sum_weight_estimate = self.calculate_total_weight_count()
+
+            # Calculate weight estimate per model
+            per_model_weight_estimate: int = (sum_weight_estimate / expert_count) if (expert_count > 0) else sum_weight_estimate
+
+            self.metadata.size_label = gguf.size_label(expert_count, per_model_weight_estimate)
```

Comment:
There are multiple problems with the parameter count which I think should be fixed before this is merge ready.

- Inaccurate MoE model parameter count
  - It's not as simple as a division of the total parameter count. You wouldn't want to call Mixtral-8x7B as Mixtral-8x5.84B, would you?
- Reading the original model files twice

This can all be fixed by using the tensor metadata from `GGUFWriter` at the right time. I'll try to implement this if you would prefer that.